web_programming/fetch_anime_and_play.py

```diff
@@ -0,0 +1,205 @@
+from xml.dom import NotFoundErr
+
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from fake_useragent import UserAgent
+from requests.exceptions import RequestException
+
+BASE_URL = "https://ww1.gogoanime2.org"
+
+
+def search_scraper(anime_name: str) -> list:
+
+    """[summary]
+
+    This function will take an url and
+    return list of anime after scraping the site.
+
+    >>> type(search_scraper("demon_slayer"))
+    <class 'list'>
+
+    Args:
+        anime_name (str): [Name of anime]
+
+    Raises:
+        e: [Raises exception on failure]
+
+    Returns:
+        [list]: [List of animes]
+    """
+
+    # concat the name to form the search url.
+    search_url = f"{BASE_URL}/search/{anime_name}"
+
+    try:
+        response = requests.get(
+            search_url, headers={"UserAgent": UserAgent().chrome}
+        )  # request the url.
+
+        # Is the response ok?
+        response.raise_for_status()
+    except RequestException as e:
+        raise e
```

Comment:
There is no need to catch an Exception if you are only going to raise the same Exception unmodified.

Reply:
ok. i've used response.raise_for_status() instead.